pkg/chunk/encoding/bigchunk.go

```diff
@@ -332,3 +336,12 @@ func firstAndLastTimes(c chunkenc.Chunk) (int64, int64, error) {
 	}
 	return first, last, iter.Err()
 }
+
+// emptyIterator has no samples.
+type emptyIterator struct{}
```

Comment:
How about using `NewNopIterator()` from TSDB instead of this?
AFAICS only `Batch()` would behave differently, and that might indicate a bug.

Reply:
`NewNopIterator()` returns a `tsdb.Iterator` as seen here: https://github.com/prometheus/tsdb/blob/719b57db44f5e5dbad7d813a5645c222c3ba7647/chunkenc/chunk.go#L55-L65

Not sure how it'll be simpler to use that than just add a few lines here.

Reply:
because then you won't add any lines here?

I'm pretty sure it returns a `chunkenc.Iterator`, i.e. it fits in `curr`.

Reply:
I created #1461 to make clear what I meant.